BungeeCord-Patches/0064-Adventure.patch

```diff
@@ -0,0 +1,1874 @@
+      * @param message the message to send
++     * @deprecated in favour of {@link Audience#sendMessage(Component)}
+      */
++    @Deprecated // Waterfall - we're going on an adventure!
+     public void sendMessage(BaseComponent message);
+ 
+     /**
+diff --git a/api/src/main/java/net/md_5/bungee/api/ProxyServer.java b/api/src/main/java/net/md_5/bungee/api/ProxyServer.java
+index a4011335..f20f401b 100644
+--- a/api/src/main/java/net/md_5/bungee/api/ProxyServer.java
++++ b/api/src/main/java/net/md_5/bungee/api/ProxyServer.java
+@@ -4,11 +4,19 @@ import com.google.common.base.Preconditions;
+ import java.io.File;
+ import java.net.InetSocketAddress;
+ import java.net.SocketAddress;
++import java.text.MessageFormat;
+ import java.util.Collection;
++import java.util.Locale;
+ import java.util.Map;
+ import java.util.UUID;
+ import java.util.logging.Logger;
+ import lombok.Getter;
++// Waterfall start - we're going on an adventure!
++import net.kyori.adventure.audience.Audience;
++import net.kyori.adventure.audience.ForwardingAudience;
++import net.kyori.adventure.text.Component;
++// Waterfall end
++import net.kyori.adventure.text.TranslatableComponent;
+ import net.md_5.bungee.api.chat.BaseComponent;
+ import net.md_5.bungee.api.config.ConfigurationAdapter;
+ import net.md_5.bungee.api.config.ServerInfo;
+@@ -16,8 +24,9 @@ import net.md_5.bungee.api.connection.ProxiedPlayer;
+ import net.md_5.bungee.api.plugin.Plugin;
+ import net.md_5.bungee.api.plugin.PluginManager;
+ import net.md_5.bungee.api.scheduler.TaskScheduler;
++import org.checkerframework.checker.nullness.qual.NonNull;
+ 
+-public abstract class ProxyServer
++public abstract class ProxyServer implements ForwardingAudience // Waterfall - We're going on an adventure!
+ {
+ 
+     @Getter
+@@ -170,9 +179,20 @@ public abstract class ProxyServer
+      * Gracefully mark this instance for shutdown.
+      *
+      * @param reason the reason for stopping. This will be shown to players.
++     * @deprecated in favour of {@link ProxyServer#stop(Component)}
+      */
++    @Deprecated // Waterfall - we're going on an adventure!
+     public abstract void stop(String reason);
+ 
++    // Waterfall start - we're going on an adventure!
++    /**
++     * Gracefully mark this instance for shutdown.
++     *
++     * @param reason the reason for stopping. This will be shown to players.
++     */
++    public abstract void stop(final @NonNull Component reason);
++    // Waterfall end
++
+     /**
+      * Register a channel for use with plugin messages. This is required by some
+      * server / client implementations.
+@@ -220,9 +240,23 @@ public abstract class ProxyServer
+      * @param motd the motd when used as a forced server
+      * @param restricted whether the server info restricted property will be set
+      * @return the constructed instance
++     * @deprecated in favour of {@link #constructServerInfo(String, InetSocketAddress, Component, boolean)}
+      */
++    @Deprecated
```

Comment:
Sometimes you comment on `@Deprecated`, other times you don't?

Reply:
Yeah sorry, that’s a mistake. That’s just a case of me forgetting about that one haha.